### PR TITLE
refactor(mobile): riverpod codegen + riverpod lint

### PR DESCRIPTION
**Pull request by shenlong-tanwen**
#### Change made in the PR

- adds `riverpod_generator` to use code generation for generating providers
- adds `riverpod_lint` and `custom_lint` to provide linting for said providers
- added an extension `ScaffoldBody` on `AsyncValue` to show the `ImmichLoadingIndicator` when the value is in loading state or a default Error widget `ScaffoldErrorBody` when an error is occurred after logging the error to ImmichLogger.
- Migrated `person.provider.dart`, `api.provider.dart` and `app_settings.provider.dart` to use code generation

### CODE REVIEW

File: mobile/lib/modules/settings/providers/app_settings.provider.dart [hunk 1]
```diff
@@ -1,4 +1,9 @@
-import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:immich_mobile/modules/settings/services/app_settings.service.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
 
-final appSettingsServiceProvider = Provider((ref) => AppSettingsService());
+part 'app_settings.provider.g.dart';
+
+@Riverpod(keepAlive: true)
+AppSettingsService appSettingsService(AppSettingsServiceRef ref) {
```
Comment (fyfrey) [inline mobile/lib/modules/settings/providers/app_settings.provider.dart:7]: can this also be a short lamba?

File: mobile/lib/shared/providers/api.provider.dart [hunk 2]
```diff
@@ -1,4 +1,9 @@
-import 'package:hooks_riverpod/hooks_riverpod.dart';
 import 'package:immich_mobile/shared/services/api.service.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
 
-final apiServiceProvider = Provider((ref) => ApiService());
+part 'api.provider.g.dart';
+
+@Riverpod(keepAlive: true)
+ApiService apiService(ApiServiceRef ref) {
```
Comment (fyfrey) [inline mobile/lib/shared/providers/api.provider.dart:7]: can this also be a short lamba?

File: mobile/lib/shared/ui/scaffold_error_body.dart [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flutter/material.dart';
+import 'package:immich_mobile/extensions/build_context_extensions.dart';
+
+// Error widget to be used in Scaffold when an AsyncError is received
+class ScaffoldErrorBody extends StatelessWidget {
+  final IconData icon;
+
+  const ScaffoldErrorBody({this.icon = Icons.ac_unit, super.key});
```
Comment (fyfrey) [inline mobile/lib/shared/ui/scaffold_error_body.dart:9]: why is the default `ac_unit`?

**Review by fyfrey — COMMENTED** [review]
I like the normal function-style for the providers :-)

File: mobile/lib/shared/providers/api.provider.dart (hunk 2 above)
Comment (shenlong-tanwen) [reply mobile/lib/shared/providers/api.provider.dart]: Oh yes, Updated 😄

File: mobile/lib/modules/settings/providers/app_settings.provider.dart (hunk 1 above)
Comment (shenlong-tanwen) [reply mobile/lib/modules/settings/providers/app_settings.provider.dart]: Updated.

File: mobile/lib/shared/ui/scaffold_error_body.dart (hunk 3 above)
Comment (shenlong-tanwen) [reply mobile/lib/shared/ui/scaffold_error_body.dart:9]: I don't really have a reason for that. The Ice Crystal icon looked good with the Error Occurred text above. I'm open for suggestions to change it though 😅

Comment (fyfrey) [reply mobile/lib/shared/ui/scaffold_error_body.dart:9]: `error_outline` as it is the generic error icon?

**Review by fyfrey — APPROVED** [review]

**Comment by alextran1502** [comment]
Please assign the PR to me when you are done with the update and it is ready to merge

**Pull request merged**